### PR TITLE
[SP-3666] Backport of PIR-1339 - Datepicker in PIR prompt incorrectly applies date format with minutes (7.1 Suite)

### PR DESCRIPTION
**Pull request by Advent51**

### CODE REVIEW

**Comment by Advent51** [comment]
@pamval @tmorgner @mchen-len-son please review

File: impl/client/src/main/javascript/web/prompting/components/DojoDateTextBoxComponent.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ define(['cdf/components/BaseComponent', "dojo/date/locale", 'dijit/form/DateText
           var dojoFormat;
           var dateFormat = this._getDateFormat();
 
-          if(dateFormat.match(/(^|(?!y).)(y{1}(?!y))/)) {
+          if(dateFormat.match(/o/)) {
```
Comment (tmorgner) [inline impl/client/src/main/javascript/web/prompting/components/DojoDateTextBoxComponent.js]: Why "o"? I cannot find "o" as valid symbol in either the [Dojo format documentation][1] nor the [SimpleDateFormat][2] documentation. What do I miss?

[1]: https://dojotoolkit.org/reference-guide/1.10/dojo/date/locale/format.html
[2]: https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html

Comment (Advent51) [inline impl/client/src/main/javascript/web/prompting/components/DojoDateTextBoxComponent.js]: @tmorgner https://api.jqueryui.com/datepicker/ 
I don't remember why, but jquery was the reason.

Comment (Advent51) [inline impl/client/src/main/javascript/web/prompting/components/DojoDateTextBoxComponent.js]: @tmorgner https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/prompting/components/DojoDateTextBoxComponent.js#L222-L235
if jquery fromat was used, then it is legacy date format

**Pull request merged**